libs/use-paystack.ts

```diff
@@ -35,42 +27,58 @@ export default function usePaystackPayment(hookConfig: HookConfig): InitializePa
       connect_account,
       connect_split,
     } = args;
+    const paystackArgs: Record<string, any> = {
+      onSuccess: onSuccess ? onSuccess : () => null,
+      onCancel: onClose ? onClose : () => null,
+      key: publicKey,
+      ref: reference,
+      email,
+      firstname,
+      lastname,
+      amount,
+      currency,
+      plan,
+      subaccount,
+      transaction_charge,
+      bearer,
+      label,
+      metadata,
```

Comment:
Why not mix the rest in like:

const paystackArgs: Record<string, any> = {
      onSuccess: onSuccess ? onSuccess : () => null,
      onCancel: onClose ? onClose : () => null,
      key: publicKey,
      ref: reference,
      email,
      firstname,
      lastname,
      amount,
      currency,
      plan,
      subaccount,
      transaction_charge,
      bearer,
      label,
      metadata,
      ...(phone && { phone }),
      ...(quantity && { quantity }),

etc etc?

Reply:
Check again!!!